packages/marshal/src/passStyleOf.js

```diff
@@ -190,6 +172,51 @@ function isPassByCopyRecord(val) {
   return true;
 }
 
+// Below we have a series of predicate functions and their (curried) assertion
+// functions. The semantics of the assertion function is just to assert that
+// the corresponding predicate function would have returned true. But it
+// reproduces the internal tests so failures can give a better error message.
+
+/**
+ * @callback Checker
+ * @param {boolean} cond
+ * @param {ReturnType<typeof X>} details
```

Comment:
The parameter of `assert` you forward to is typed
```suggestion
 * @param {Details=} details
```
which is different in two ways.
   * `ReturnType<typeof X>` is `DetailsTag`, where `Details` is either that or a string.
   * The parameter of `assert` is optional.

Also, I'm curious about the pros and cons of the `ReturnType` style vs using the type name directly. Thoughts?

Reply:
I couldn't figure out how to extract the `Details` type from `@agoric/assert`, because it doesn't export its types in a way that JS can use them.

I tried harder and got it now.  Still ugly, but it works:

```js
/**
 * @param {Parameters<typeof assert>[1]} details
 */
```

Reply:
How should it export its types so JS can use them? I certainly intended to export those types.

Reply:
> How should it export its types so JS can use them? I certainly intended to export those types.

I will take a look.  I'm not sure what's preventing that right now.

Reply:
Found it!  I will include a fix in a future PR and tag you on it.

Reply:
(Turned out I needed to `import '@agoric/assert/exported';` in `passStyleOf.js`, and create `packages/assert/exported.js`).